src/main/java/umaikaze/duke/task/Deadline.java

```diff
@@ -0,0 +1,27 @@
+package umaikaze.duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDateTime by;
+    protected boolean hasTime;
+
+    public Deadline(String description, LocalDateTime by, boolean hasTime) {
+        super(description);
+        this.by = by;
+        this.hasTime = hasTime;
+    }
+
+    @Override
+    public String getSaveString() {
+        return "D|" + (isDone ? 1 : 0) + "|" + description + "|"
```

Comment:
Is it nicer to make use of the parent method `getSaveString()` in this children method and simplify your codes? 😉